var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1343,6 +1344,9 @@ def setup_dependent_run_environment(
         """Set PYTHONPATH to include the site-packages directory for the
         extension and any other python extensions it depends on.
         """
+        if self.spec.satisfies("platform=windows"):
+            env.prepend_path("PATH", dependent_spec.prefix.scripts)
+            env.prepend_path("PATH", self.spec.prefix)
```

Comment:
(question) it looks like this is also preparing for `py-*` packages that put executables in the base prefix: is that something only `py-*` packages (on Windows) do?

Reply:
> is that something only py-* packages (on Windows) do?

Sorry, forgot to address this part. AFAIK Python doesn't install any tooling into the base prefix on non Windows platforms (I don't think there's as much peripheral tooling provided by non Windows in general as the tooling is usually to overcome Windows hurdles)

Reply:
> setup_run_environment

Is run environment established for build dep types?

Reply:
All `PythonPackage`s `extend` `python`, which imposes a `build/run` dependency on it, so they should always run `setup_run_environment` from `python` (the `build_environment` logic does that for `BUILDTIME_DIRECT` dependencies).

Reply:
Ah ok, I didn't realize `extend` had that effect, thanks! 
For packages that depend on, but do not extend python (icu4c for example), we'd still need the base prefix available during build time however.

Reply:
`icu4c` declares a build dependency on python (and in general anything that needs python exes at build time should do that).

Reply:
Ahh, wait, I realized you're only asking me to remove this from setup__dependent__run_environment, not build 🤦 

Sorry for the churn, edit made and pushed.